src/app/cli/src/init/client.ml

```diff
@@ -2169,6 +2169,216 @@ let thread_graph =
+                (Error.to_string_hum err) ;
+              exit 1
+          | Error err ->
+              Format.eprintf "Failed to get fee payer balance, error: %s@."
+                (Error.to_string_hum err) ;
+              exit 1
+        in
+        let fee_payer_balance_as_amount =
+          Currency.Balance.to_amount fee_payer_balance
+        in
+        let%bind fee_payer_initial_nonce =
+          (* inferred nonce considers txns in pool, in addition to ledger *)
+          match%bind
+            get_nonce ~rpc:Daemon_rpcs.Get_inferred_nonce.rpc
+              fee_payer_account_id port
+          with
+          | Ok nonce ->
+              Deferred.return @@ Account.Nonce.of_uint32 nonce
+          | Error err ->
+              Format.eprintf "Failed to get fee payer nonce: %s@." err ;
+              exit 1
+        in
+        Format.printf "Fee payer public key: %s, initial nonce = %d@."
+          ( Account_id.public_key fee_payer_account_id
+          |> Public_key.Compressed.to_base58_check )
+          (Account.Nonce.to_int fee_payer_initial_nonce) ;
+        let fee_payer_current_nonce = ref fee_payer_initial_nonce in
+        let keys_per_zkapp = 8 in
+        let keypairs =
+          List.init num_accounts ~f:(fun _n -> Signature_lib.Keypair.create ())
+        in
+        let keypair_chunks = List.chunks_of keypairs ~length:keys_per_zkapp in
+        let num_chunks = List.length keypair_chunks in
+        (* amount + fees must not be more than fee payer balance *)
+        let amount_and_fees =
+          let total_fees = num_chunks * fee in
+          Currency.Amount.of_nanomina_int_exn (amount + total_fees)
+        in
+        if Currency.Amount.( > ) amount_and_fees fee_payer_balance_as_amount
+        then (
+          Format.eprintf
+            !"Amount plus fees (%{sexp: Currency.Amount.t}) is greater than \
+              fee payer balance (%{sexp: Currency.Amount.t})@."
+            amount_and_fees fee_payer_balance_as_amount ;
+          Core.exit 1 ) ;
+        let amount_per_chunk = amount / num_chunks in
+        let amount_rem = amount mod num_chunks in
+        (* add 1 to enough per-chunk amounts so total over all chunks is amount *)
+        let chunk_amounts =
+          Array.init num_chunks ~f:(fun i ->
+              if i < amount_rem then amount_per_chunk + 1 else amount_per_chunk )
+        in
+        assert (Array.fold chunk_amounts ~init:0 ~f:( + ) = amount) ;
+        Deferred.List.iteri keypair_chunks ~f:(fun i kps ->
+            let chunk_amount = chunk_amounts.(i) in
+            let num_updates = List.length kps in
+            let amount_per_update = chunk_amount / num_updates in
+            let update_rem = chunk_amount mod num_updates in
+            let update_amounts =
+              Array.init num_updates ~f:(fun i ->
+                  if i = 0 then
+                    if 0 < amount_rem then amount_per_update + 1 - fee
```

Comment:
> We could change that calculation, by spreading the fee deduction over all the updates, but there will still be some variation, if the fee isn't evenly divisible by the number of receivers.

Yeah, that's fine

Reply:
Made that change, so that all account updates within a chunk of up-to-8 keys will get the same amount (within 1 nanomina).

Each chunk gets the same amount (within 1 nanomina). That means that a straggler chunk of fewer than 8 keys will get more per key. I can even that out by distributing the amount evenly over all keys.